components/calendar/nz-calendar.component.ts

```diff
@@ -108,22 +113,26 @@ export class NzCalendarComponent implements ControlValueAccessor, OnInit {
     this.calculateActiveMonth();
   }
 
-  onModeChange(mode: 'month'|'year'): void {
+  onModeChange(mode: ModeType): void {
     this.nzModeChange.emit(mode);
+    this.nzPanelChange.emit({'date': this.activeDate, 'mode': mode});
   }
 
   onDateSelect(date: Date): void {
     this.updateDate(date);
+    this.nzSelectChange.emit(date);
   }
 
   onYearSelect(year: number): void {
     const date = setYear(this.activeDate, year);
     this.updateDate(date);
+    this.nzPanelChange.emit({'date': date, 'mode': this.nzMode});
   }
 
   onMonthSelect(month: number): void {
     const date = setMonth(this.activeDate, month);
     this.updateDate(date);
+    this.nzPanelChange.emit({'date': date, 'mode': this.nzMode});
```

Comment:
should be this.nzSelectChange.emit(date);

Reply:
I have fixed it & modified nz-calendar.spec.ts